[MC-1243] Manage Engine Service Desk 1.0.0 (#1390)
rapid7/insightconnect-plugins: plugins/manage_engine_service_desk/help.md

* If you want to generate the API key for a new technician, click Add New Technician link, enter the technician details and provide login permission.
* Click Generate link under the API key details block. You can select a time frame for the key to expire using the calendar icon or simply retain the same key indefinitely.
* If a key is already generated for the technician, a Re-generate link appears.

# Supported Product Versions

* ServiceDesk Plus 13008

# Documentation

## Setup

The connection configuration accepts the following parameters:

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|api_key|credential_secret_key|None|True|Manage Engine Service Desk Technican's API key|None|EXAMPLE1-API2-KEY3-HDFS-48GS24WSA6GE|
|sdp_base_url|string|None|True|Service Desk Plus Base URL|None|https://example.com|
|ssl_verify|boolean|False|True|SSL verify|None|False|

Example input:

```
{
  "api_key": "EXAMPLE1-API2-KEY3-HDFS-48GS24WSA6GE",
  "sdp_base_url": "http://me-sdeskplus.dev.example.com:8080",
  "ssl_verify": false
}
```

## Technical Details

### Actions

#### Add Request

> This action allows you to add new a request. Subject and requester parameters are required, others are optional. In every parameter containing `ID` and `Name` fields please provide only one of them.

Reviewer: ```suggestion
Add a new request. Subject and requester parameters are required, others are optional. In every parameter containing `ID` and `Name` fields please provide only one of them.
```

|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 55,
  "request_note_id": 336,
  "status": "success",
  "status_code": 2000
}
```

#### Add Resolution

Add or update the resolution of a request.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|request_id|integer|None|True|The id of the request|None|27|
|content|string|None|True|Resolution content|None|Sample resolution|
|add_to_linked_requests|boolean|None|True|Whether the resolution should be added to linked requests|None|True|

Example input:

```
{
  "add_to_linked_requests": true,
  "content": "Sample resolution",
  "request_id": 27
}
```

##### Output

|Name|Type|Required|Description|Example|
|----|----|--------|-----------|--------|
|request_id|integer|False|The id of the request|27|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 27,
  "status": "success",
  "status_code": 2000
}
```

#### Assign Request

> This action allows you to assign a request to a technician or group. Request ID is required, as well as at least one of Group or Technician. In every parameter containing `ID` and `Name` fields please provide only one or the other.

Reviewer: ```suggestion
Assign a request to a technician or group. Request ID is required, as well as at least one of Group or Technician. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```

Example input:

```
{
  "request_id": 54
}
```

##### Output

|Name|Type|Required|Description|Example|
|----|----|--------|-----------|--------|
|request_id|integer|False|The id of deleted request|54|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 54,
  "status": "success",
  "status_code": 2000
}
```

#### Delete Request Note

Delete a given request note on a specific request.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|request_id|integer|None|True|The id of the request|None|55|
|request_note_id|integer|None|True|The id of the request note to delete|None|208|

Example input:

```
{
  "request_id": 55,
  "request_note_id": 208
}
```

##### Output

|Name|Type|Required|Description|Example|
|----|----|--------|-----------|--------|
|request_id|integer|False|The id of the request|55|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 55,
  "status": "success",
  "status_code": 2000
}
```

#### Edit Request

> This action allows you to update the request by using the unique request ID. At least one parameter other than Request ID is required. In every parameter containing `ID` and `Name` fields please provide only one or the other.

Reviewer: ```suggestion
Update the given request. At least one parameter other than Request ID is required. In every parameter containing `ID` and `Name` fields please provide only one or the other.
```

    },
    "subcategory": {
      "name": "MS Office",
      "id": 5
    },
    "email_ids_to_notify": [
      "user@example.com"
    ],
    "udf_fields": {},
    "created_time": "Aug 3, 2022 01:11 AM",
    "due_by_time": "Aug 3, 2022 01:00 PM",
    "created_by": {
      "id": 4,
      "name": "administrator",
      "is_vipuser": false
    },
    "is_service_request": false,
    "has_notes": false,
    "is_overdue": true
  },
  "status": "success"
}
```

#### Get Resolution

Get the resolution of the given request.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|request_id|integer|None|True|The id of the request|None|27|

Example input:

```
{
  "request_id": 27
}
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|request_id|integer|False|The id of the request|27|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|
|content|string|False|Resolution content|Sample resolution|

Example output:

```
{
  "request_id": 27,
  "content": "asd",
  "status": "success",
  "status_code": 2000
}
```

#### Pickup Request

> This action allows you to pick up a request in your name as a technician.

Reviewer: ```suggestion
Pick up (assign) a given request in your name as a technician.
```

|description|string|None|False|Note description in HTML format|None|Need help|
|mark_first_response|boolean|None|False|Whether to set the responded date of the request/ticket|None|True|
|notify_technician|boolean|None|False|Whether to notify the technician or not|None|True|
|request_id|integer|None|True|The id of the request|None|55|
|request_note_id|integer|None|True|The id of the request note|None|209|
|show_to_requester|boolean|None|False|Whether to show the note to requester or not|None|False|

Example input:

```
{
  "add_to_linked_request": false,
  "description": "Need help",
  "mark_first_response": true,
  "notify_technician": true,
  "request_id": 55,
  "request_note_id": 209,
  "show_to_requester": false
}
```

##### Output

|Name|Type|Required|Description|Example|
|----|----|--------|-----------|--------|
|request_id|integer|False|The id of the request|55|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 55,
  "status": "success",
  "status_code": 2000
}
```

#### Get List Request

> This action lets you view the details of all the requests.

Reviewer: ```suggestion
View the details of a list of requests matching a search.
```

        "is_vipuser": false
      },
      "site": {
        "name": "Test Site",
        "id": 301
      },
      "status": {
        "name": "Open",
        "id": 2
      },
      "subject": "Plugin action test add request 2",
      "technician": {
        "id": 4,
        "name": "administrator",
        "is_vipuser": false
      }
    },
    {
      "created_by": {
        "id": 4,
        "name": "administrator",
        "is_vipuser": false
      },
      "created_time": "Jul 13, 2022 01:20 AM",
      "due_by_time": "Jul 15, 2022 05:00 PM",
      "id": 27,
      "is_overdue": true,
      "is_service_request": false,
      "priority": {
        "name": "Low",
        "id": 1
      },
      "requester": {
        "id": 4,
        "name": "administrator",
        "is_vipuser": false
      },
      "site": {
        "name": "Test Site",
        "id": 301
      },
      "status": {
        "name": "Open",
        "id": 2
      },
      "subject": "Plugin action test add request 2",
      "technician": {
        "id": 4,
        "name": "administrator",
        "is_vipuser": false
      }
    }
  ],
  "status": "success"
}
```

#### Get List Request Notes

> This action allows you to get all the notes.

Reviewer: ```suggestion
Get the list of all notes associated with the given request.
```

Example input:

```
{
  "request_id": 55
}
```

##### Output

|Name|Type|Required|Description|Example|
|----|----|--------|-----------|--------|
|request_id|integer|True|The id of the request|55|
|notes|[]note|False|Notes assigned to the request|["{"note_id": "312", "added_time": "Jul 8, 2022 02:02 AM", "added_by": {"name": "John", "id": "71"}}"]|
|status|string|True|Status of the request|success|
|status_code|integer|False|Status code of the request|2000|

Example output:

```
{
  "request_id": 55,
  "notes": [
    {
      "last_updated_by": {
        "name": "administrator",
        "id": 4
      },
      "added_time": "Aug 3, 2022 12:55 AM",
      "last_updated_time": "Aug 3, 2022 12:55 AM",
      "added_by": {
        "name": "administrator",
        "id": 4
      },
      "show_to_requester": true
    },
    {
      "last_updated_by": {
        "name": "administrator",
        "id": 4
      },
      "added_time": "Aug 3, 2022 12:55 AM",
      "last_updated_time": "Aug 3, 2022 12:55 AM",
      "added_by": {
        "name": "administrator",
        "id": 4
      },
      "show_to_requester": true
    }
  ],
  "status": "success",
  "status_code": 2000
}
```

#### Get Request

> This action allows you to view the request details by the unique request id.

Reviewer: ```suggestion
View the details of a request given the request id.
```

    },
    "status": {
      "name": "Open",
      "id": 2
    },
    "mode": {
      "name": "Web Form",
      "id": 2
    },
    "level": {
      "name": "Tier 4",
      "id": 4
    },
    "urgency": {
      "name": "Urgent",
      "id": 1
    },
    "priority": {
      "name": "Normal",
      "id": 2
    },
    "service_category": {
      "name": "User Management",
      "id": 8
    },
    "id": 54,
    "assets": [
      {
        "name": "keyboardMC",
        "id": 301
      }
    ],
    "site": {
      "name": "Test Site",
      "id": 301
    },
    "technician": {
      "id": 4,
      "name": "administrator",
      "is_vipuser": false
    },
    "category": {
      "name": "Software",
      "id": 3
    },
    "subcategory": {
      "name": "MS Office",
      "id": 5
    },
    "email_ids_to_notify": [
      "user@example.com"
    ],
    "udf_fields": {},
    "created_time": "Aug 3, 2022 01:11 AM",
    "due_by_time": "Aug 3, 2022 01:00 PM",
    "created_by": {
      "id": 4,
      "name": "administrator",
      "is_vipuser": false
    },
    "is_service_request": false,
    "has_notes": false,
    "is_overdue": true
  },
  "status": "success"
}
```

#### Get Resolution

> This action allows you to get a resolution of the request.

Reviewer: ```suggestion
Get the resolution of the given request.
```
